gc.c

```diff
@@ -3131,13 +3131,7 @@ obj_free(rb_objspace_t *objspace, VALUE obj)
             rb_id_table_free(RCLASS_CVC_TBL(obj));
 	}
 	if (RCLASS_SUBCLASSES(obj)) {
```

Comment:
`rb_class_remove_subclass_head` will already check that `RCLASS_SUBCLASSES` is present. We don't need to check it here.